### ISSUE TITLE
Only commented SQL with L052:require_final_semicolon = True wrongly reports a missing ending semi-column

### ISSUE THREAD

**Issue by niconoe-** [sqlfluff/sqlfluff#4435]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In my context, I need to run SQLFluff on lots of SQL files. For each SQL file, if there's a statement, I absolutely need a final semicolon at the end of that statement, so I configured SQLFluff to run the rule L052 with `require_final_semicolon: True`. But it may happen in some cases that my SQL files are only containing commented snippets, and there are 0 statements in such file. Because of the presence of this option in the rule L052, I have an error on my file.

### Expected Behaviour

```
All Finished!     
```

### Observed Behaviour

```
== [99999999999-test.sql] FAIL
L:   1 | P:   1 | L052 | Statements must end with a semi-colon.
All Finished!
```

### How to reproduce

In order to reproduce, just run SQLFluff on a file that contains only commented snippets like
```sql
-- SELECT * FROM my_table;

/*
INSERT INTO my_table VALUES 
('foo', 'FOO')
('bar', 'BAR')
*/
```

But **be sure you are using the following configuration overload**:
```config
[sqlfluff:rules:L052]
require_final_semicolon = True
```

### Dialect

MySQL, but I think this issue is agnostic.

### Version

I'm using the latest version of SQLFluff, as I'm using the Dockerimage `sqlfluff/sqlfluff:latest`

### Configuration

[sqlfluff:rules:L052]
require_final_semicolon = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by james-johnston-thumbtack** [comment]
Even an empty file with only whitespace will trigger this!
